nixos/modules/services/computing/slurm/slurm.nix

```diff
@@ -0,0 +1,484 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.slurm;
+  # configuration file can be generated by http://slurm.schedmd.com/configurator.html
+  configFile = pkgs.writeText "slurm.conf" 
+    ''
+      ${optionalString (cfg.controlMachine != null) ''controlMachine=${cfg.controlMachine}''}
+      ${optionalString (cfg.controlAddr != null) ''controlAddr=${cfg.controlAddr}''}
+      ${optionalString (cfg.authType != null) ''authType=${cfg.authType}''}
+      ${optionalString (cfg.cacheGroups != null) ''cacheGroups=${cfg.cacheGroups}''}
+      ${optionalString (cfg.cryptoType != null) ''cryptoType=${cfg.cryptoType}''}
+      ${optionalString (cfg.mpiDefault != null) ''mpiDefault=${cfg.mpiDefault}''}
```

Comment:
You need a submodule to have `services.slurm.partition.<name>`. Then use map and concatenate strings. Then again I wonder if that makes sense to have as option? It should be fine with extraConfig as it's just lines, no?